xbmc/Util.cpp

```diff
@@ -1834,18 +1820,41 @@ void CUtil::ScanForExternalSubtitles(const std::string& strMovie, std::vector<st
     || item.IsLiveTV()
     || !item.IsVideo())
     return;
-  
+
+  CLog::Log(LOGDEBUG,"%s: Searching for subtitles...", __FUNCTION__);
+
+  std::string strBasePath;
+  std::string strSubtitle = URIUtils::ReplaceExtension(URIUtils::GetFileName(strMovie), "");
+
+  if (item.HasVideoInfoTag())
+    strBasePath = item.GetVideoInfoTag()->m_basePath;
+
+  if (item.IsOpticalMediaFile())
+    strBasePath = item.GetLocalMetadataPath();
+
+  if (strBasePath.empty())
+    strBasePath = URIUtils::GetBasePath(strMovie);
+
+  CFileItemList items;
   vector<std::string> strLookInPaths;
-  
-  std::string strMovieFileName;
-  std::string strMoviePath;
-  
-  URIUtils::Split(strMovie, strMoviePath, strMovieFileName);
-  std::string strMovieFileNameNoExt(URIUtils::ReplaceExtension(strMovieFileName, ""));
-  strLookInPaths.push_back(strMoviePath);
-  
+
+  int flags = DIR_FLAG_NO_FILE_DIRS | DIR_FLAG_READ_CACHE | DIR_FLAG_NO_FILE_INFO;
+
+  if (!strBasePath.empty())
+    CDirectory::GetDirectory(strBasePath, items, g_advancedSettings.m_subtitlesExtensions, flags);
+
+  const char * common_sub_dirs[] = {"subs", "subtitles", "vobsubs", "sub", "vobsub", "subtitle", NULL};
+
+  for (int i = 0; i < items.Size(); ++i)
+  {
+    for (int j = 0; common_sub_dirs[j]; j++)
+    {
+      if (StringUtils::EqualsNoCase(items[i]->GetLabel(), common_sub_dirs[j]))
+        strLookInPaths.push_back(items[i]->GetPath());
+    }
+  }
+
   CURL url(strMovie);
-  std::string isoFileNameNoExt;
   if (url.IsProtocol("bluray"))
       url = CURL(url.GetHostName());
```

Comment:
Not my indent ;) But i will change anyway and probably get rid of the entire block.